test-data/unit/check-namedtuple.test

```diff
@@ -973,3 +973,58 @@ B = namedtuple('X', ['a'])         # E: First argument to namedtuple() should be
 C = NamedTuple('X', [('a', 'Y')])  # E: First argument to namedtuple() should be "C", not "X"
 class Y: ...
 [builtins fixtures/tuple.pyi]
+
+[case testNamedTupleTypeIsASuperTypeOfOtherNamedTuples]
+from typing import Tuple, NamedTuple
+
+class Bar(NamedTuple):
+    name: str = "Bar"
+
+class Baz(NamedTuple):
+    a: str
+    b: str
+
+def print_namedtuple(obj: NamedTuple) -> None:
+    pass
```

Comment:
Could you add some tests that verify that you can actually do something useful with a parameter associated as NamedTuple? For example, call `_asdict()` as in the bug report.

For what it's worth I'm skeptical that this is useful. NamedTuple is not itself a type and it doesn't make sense to me to use it as a type.

Also unrelatedly, thank you for all your contributions to mypy! It's been great seeing so many improvements.

Reply:
> Could you add some tests that verify that you can actually do something useful with a parameter associated as NamedTuple?

Sure! Will do.

> For what it's worth I'm skeptical that this is useful. NamedTuple is not itself a type and it doesn't make sense to me to use it as a type.

It can be used on a lower level for building validation libraries / a-la-pydantic / serialization tools. Something like https://github.com/dry-python/classes/

Since `NamedTuple` has several attrs / methods that can be useful in this context: `._as_dict()`, `_fields`, etc.
Another option is to define a protocol with the same props, but it looks like that we can save our users from it with a simple tweak.

> Also unrelatedly, thank you for all your contributions to mypy! It's been great seeing so many improvements.

Thanks! 😊

Reply:
Done, added `._source`